subprojects/core-api/src/main/java/org/gradle/api/provider/Property.java

```diff
@@ -77,29 +105,47 @@
 
     /**
-     * Specifies the value to use as the convention for this property. The convention is used when no value has been set for this property.
+     * Specifies the value to use as the convention (default value) for this property. If the convention is set and
+     * no explicit value or value provider has been specified, then the convention will be returned as the value of
+     * the property (when queried).
      *
-     * @param value The value, or {@code null} if the convention is that the property should have no value.
+     * <p>This method can be used to specify that the property does not have a default value, by passing it
+     * {@code null}.</p>
+     *
+     * @param value The convention value, or {@code null} if the property should have no default value.
      * @return this
      * @since 5.1
      */
     Property<T> convention(@Nullable T value);
 
     /**
-     * Specifies the provider of the value to use as the convention for this property. The convention is used when no value has been set for this property.
+     * Specifies the provider to be used to query the convention (default value) for this property. If a convention
+     * provider has been set and no explicit value or value provider has been specified, then the convention
+     * provider's value will be returned as the value of the property (when queried).
+     *
+     * <p>The property's convention tracks the convention provider. Whenever the convention's actual value is
+     * needed, the convention provider will be queried anew.</p>
+     *
+     * <p>This method can't be used to specify that a property does not have a default value. Passing in a {@code null}
+     * provider will result in an {@code IllegalArgumentException} being thrown.</p>
      *
-     * @param valueProvider The provider of the value.
+     * @param provider The provider of the property's convention value, can't be null.
      * @return this
      * @since 5.1
      */
-    Property<T> convention(Provider<? extends T> valueProvider);
+    Property<T> convention(Provider<? extends T> provider);
 
     /**
-     * Disallows further changes to the value of this property. Calls to methods that change the value of this property, such as {@link #set(Object)} or {@link #set(Provider)} will fail.
+     * Disallows further changes to the value of this property. Calls to methods that change the value of this property,
+     * such as {@link #set(Object)}, {@link #set(Provider)}, {@link #value(Object)} and {@link #value(Provider)} will fail,
+     * by throwing an {@code IllegalStateException}.
      *
-     * <p>When this property has a value provided by a {@link Provider}, the value of the provider is queried when this method is called and the value of this property set to the result. The value of the provider will no longer be tracked.</p>
+     * <p>When this property's value is specified via a {@link Provider}, this method call will trigger the querying
```

Comment:
I would say `calling finalizeValue will trigger...` instead of `this method call will trigger` to avoid any ambiguity of what method is called.